app/services/message_director.rb

```diff
@@ -3,78 +3,90 @@
-      preferred_channel = HighlandsSSO.preferred_channel(receiver)
-      case preferred_channel
-      when :email
-        message = Sendgrid.send(message_vo)
-        if message.error
-          msg = message.error[:error]
-          log_error(msg)
-        else
-          msg = "Sent Email: subject (#{message_vo.email_subject}) from (#{message_vo.manager_name}) to (#{message_vo.receiver_email})"
-          log_info(msg)
-        end
-      when :sms
-        message = Clearstream.send(message_vo)
-        if message.error
-          msg = message.error[:error]
-          log_error(msg)
-        else
-          msg = "Sent SMS: subject (#{message_vo.email_subject}) from (#{message_vo.manager_name}) to (#{message_vo.receiver})"
-          log_info(msg)
-        end
-      else
-        msg = "Sent message: subject (#{message_vo.sms_message}) to (#{message_vo.receiver_email}), but receiver prefers neither Email nor SMS!"
+    when 'Clearstream'
+      message = Clearstream.send(message_vo)
+      if message.error
+        msg = message.error[:error]
         log_error(msg)
-        message = ReturnVo.new({value: nil, error: return_error(msg, :precondition_failed)})
+      else
+        msg = "Sent SMS: subject (#{message_vo.email_subject}) from (#{message_vo.manager_name}) to (#{message_vo.receiver_email})"
+        log_info(msg)
       end
-
-      message
-
-    rescue => e
-      msg = "Exception (#{e.message}) occurred in create_message for #{message_vo.to_json}"
+    else
+      msg = "Sent message: subject (#{message_vo.sms_message}) to (#{message_vo.receiver_email}), but receiver prefers neither Email nor SMS!"
       log_error(msg)
-      ReturnVo.new({value: nil, error: error_json = return_error(msg, :unprocessable_entity)})
+      message = ReturnVo.new({value: nil, error: return_error(msg, :precondition_failed)})
     end
+    # Return message in a ReturnVo
+    message
   end
 
   private
 
+  # This is where we verify that the data passed matches with data in the database and set the message target.
   def self.create_message(message_vo)
+    errs = []
     manager_id = message_vo.manager_id   # A manager is the sending app.
     manager_email = message_vo.manager_email
-    receiver = Receiver.find_by(receiver: message_vo.receiver)
-    team = Team.find_by(x_team_id: message_vo.team)
+    receiver = Receiver.find_by(receiver_sso_id: message_vo.receiver_sso_id)
+    # Currently, we only support Sendgrid (for emails) and Clearstream (for sms)
+    if receiver && receiver.preferences
+      delivery_target = receiver.preferences['email'] ? 'Sendgrid' : 'Clearstream'
+      msg_target = MsgTarget.find_by(name: delivery_target)
+      if msg_target
+        msg_target_id = msg_target.id
+      else
+        errs <<  "Invalid receiver.preferences (#{receiver.preferences}). Unable to determine message target (Email/SMS)."
+      end
+    else
+      errs << "Null receiver.preferences. Unable to determine message target (Email/SMS)."
```

Comment:
Can this variable be called `errors`? I think it is more idiomatic.

Reply:
ok